uint32 grab_or_release
---
bool success
---
uint32 step